test/specs/smoke.test.js

```diff
@@ -0,0 +1,64 @@
+const {
+  HomePage,
+  LoginPage,
+  ProductPage,
+  ShippingPage,
+  PaymentsPage,
+  SuccessPage,
+} = require("../pageobjects/pageObjects");
+const testData = require("../data/testData");
+
+describe("Smoke Test - Product Purchase Flow", () => {
+  it("should log into a valid account", async () => {
+    await HomePage.open();
+    await HomePage.navigateToLoginPage();
+
+    await LoginPage.login(testData.user.email, testData.user.password);
+
+    const welcomeMessageExists = await HomePage.welcomeMessage.isExisting();
+    expect(welcomeMessageExists).toBe(true);
+  });
+
+  it("should search for a product and open product details", async () => {
+    await HomePage.searchForProduct(testData.product.searchTerm);
+
+    await HomePage.openProductDetail(0);
+    const currentUrl = await browser.getUrl();
+    expect(currentUrl).toContain(testData.product.searchTerm);
+  });
+
+  it("should select product options and add product to cart", async () => {
+    const { size, color, quantity } = testData.product;
+    
+    await ProductPage.productTitle.waitForDisplayed();
+    await ProductPage.selectProductSize(size);
+    await ProductPage.selectProductColor(color);
+    await ProductPage.setProductQuantity(quantity);
+    await ProductPage.addToCart();
+
+    const successMessage = await ProductPage.isProductAddedToCart();
+    expect(successMessage).toContain("You added");
```

Comment:
Can this check be added to isProductAddedToChart() method?

Reply:
Change made as suggested to simplify the test. isProductAddedToCart() now returns true if the success message is present, making the test cleaner.